src/renderer/components/dock/logs/__test__/download-logs.test.tsx

```diff
@@ -0,0 +1,186 @@
+      showTimestamps: false,
+    }),
+    getPodById: (id) => {
+      if (id === selectedPod.getId()) {
+        return selectedPod;
+      }
+
+      return undefined;
+    },
+    getTimestampSplitLogs: () => [...logs],
+    getLogs: () => ["some-logs"],
+    getLogsWithoutTimestamps: () => ["some-logs"],
+    ...deps,
+  });
+};
+
+describe("download logs options in pod logs dock tab", () => {
+  let builder: ApplicationBuilder;
+  let openSaveFileDialogMock: jest.MockedFunction<() => void>;
+
+  beforeEach(() => {
+    builder = getApplicationBuilder();
+
+    builder.setEnvironmentToClusterFrame();
+
+    builder.beforeApplicationStart(({ rendererDi }) => {
+      rendererDi.override(callForAllLogsInjectable, () => () => Promise.resolve("all-logs"));
+      rendererDi.override(logsViewModelInjectable, () => getOnePodViewModel("foobar"));
+      rendererDi.override(getRandomIdForPodLogsTabInjectable, () =>
+        jest
+          .fn(() => "some-irrelevant-random-id")
+          .mockReturnValueOnce("some-first-tab-id")
+          .mockReturnValueOnce("some-second-tab-id"),
+      );
+
+      openSaveFileDialogMock = jest.fn();
+      rendererDi.override(openSaveFileDialogInjectable, () => openSaveFileDialogMock);
+    });
+
+  });
+
+  describe("when opening pod logs", () => {
+    let rendered: RenderResult;
+    let rendererDi: DiContainer;
+    
+    beforeEach(async () => {
+      rendered = await builder.render();
+      rendererDi = builder.dis.rendererDi;
+
+      const pod = dockerPod;
+      const createLogsTab = rendererDi.inject(createPodLogsTabInjectable);
+      const container = {
+        name: "docker-exporter",
+        image: "docker.io/prom/node-exporter:v1.0.0-rc.0",
+        imagePullPolicy: "pull",
+      };
+
+      const dockStore = rendererDi.inject(dockStoreInjectable);
+
+      dockStore.closeTab("terminal");
+
+      createLogsTab({
+        selectedPod: pod,
+        selectedContainer: container,
+      });
+    });
+
+    it("renders", () => {
+      expect(rendered.baseElement).toMatchSnapshot();
+    });
+
+    it("contains download dropdown button", () =>  {
+      expect(rendered.getByTestId("download-logs-dropdown")).toBeInTheDocument();
+    });
+
+    describe("when clicking on button", () => {
+      beforeEach(() => {
+        const button = rendered.getByTestId("download-logs-dropdown");
+
+        act(() => button.click());
+      });
+
+      it("shows download visible logs menu item", () => {
+        expect(rendered.getByTestId("download-visible-logs")).toBeInTheDocument();
+      });
+
+      it("shows download all logs menu item", () => {
+        expect(rendered.getByTestId("download-all-logs")).toBeInTheDocument();
+      });
+
+      it("when selected 'download visible logs', shows save dialog with proper attributes", async () => {
+        const downloadMenuItem = rendered.getByTestId("download-visible-logs");
+
+        downloadMenuItem.click();
+        
+        await waitFor(() =>
```

Comment:
`waitFor` is not good. E.g. it causes test to not fail before timeout if `openSavefileDialog` is not called at all. 

I bet that changing `callForAllLogsMock` to be `asyncFn` ja then calling `callForAllLogsMock.resolve('some-logs')` somewhere in the tests would be enough to get rid of this. 

If you still prefer using the `Promise.resolve(...)` type mock for it, then `await flushPromises()` should be enough so that you can get rid of `waitFor`.